pkg/storage/tsdb/users_scanner.go

```diff
@@ -38,6 +38,20 @@ func NewUsersScanner(bucketClient objstore.Bucket, isOwned func(userID string) (
 //
 // If sharding is enabled, returned lists contains only the users owned by this instance.
 func (s *UsersScanner) ScanUsers(ctx context.Context) (users, markedForDeletion []string, err error) {
+	deletedUsers := make(map[string]struct{})
+
+	// Discovering the deleted users from the global markers directory.
+	err = s.bucketClient.Iter(ctx, util.GlobalMarkersDir, func(entry string) error {
+		// entry will be of the form __markers__/<user>/
+		parts := strings.Split(entry, objstore.DirDelim)
+		user := parts[1]
+		deletedUsers[user] = struct{}{}
```

Comment:
Should we make sure that this is a deletion marker? Maybe in the future we may have other markers here?